### PR TITLE
Making mongoose models

### PR DESCRIPTION
**Pull request by Kobe16**
Also added in Avery's interfaces because they weren't in my version of the code. 

I referenced the priceHistoryModel inside the PokemonCardSchema because that is how our interfaces were kind of set up. Not sure if I did it right though. 

Some resources I used: 
https://tomanagle.medium.com/strongly-typed-models-with-mongoose-and-typescript-7bc2f7197722
https://www.geeksforgeeks.org/datatypes-in-mongodb/
https://stackoverflow.com/questions/18001478/referencing-another-schema-in-mongoose
https://mongoosejs.com/docs/populate.html

### CODE REVIEW

File: api/src/models/priceHistory.model.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { model, Schema, Document } from 'mongoose';
+import { PriceHistoryData } from '@interfaces/priceHistory.interface'; 
+
+const PriceHistorySchema: Schema = new Schema({
+// Will potentially calculate the date with BE logic
```
Comment (jbrw1984) [inline api/src/models/priceHistory.model.ts]: You'll need this in the schema, regardless. My feedback here was in reference to the DTO.

**Review by jbrw1984 — CHANGES_REQUESTED** [review]
I'd flush out the PriceHistorySchema

File: api/src/models/priceHistory.model.ts (hunk 1 above)
Comment (Kobe16) [reply api/src/models/priceHistory.model.ts]: Sounds good. Just added those fields back into the code

File: api/src/models/priceHistory.model.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { model, Schema, Document } from 'mongoose';
+import { PriceHistoryData } from '@interfaces/priceHistory.interface'; 
+
+const PriceHistorySchema: Schema = new Schema({
+  month: {
```
Comment (jbrw1984) [inline api/src/models/priceHistory.model.ts:5]: Approving this PR, however, typically what you will see is that M/D/Y is usually a format that can be done with either a date or time object. https://www.mongodb.com/docs/manual/reference/method/Date/

**Review by jbrw1984 — APPROVED** [review]

**Pull request merged**